earthdata/store.py

```diff
@@ -2,132 +2,266 @@
-            )
-            return session
+            ),
+            "trust_env": True,
+            "cookies": req.cookies.get_dict(),
+        }
+        if bearer_token and self.auth.authenticated:
+            client_kwargs["headers"] = {
+                "Authorization": f'Bearer {self.auth.token["access_token"]}'
+            }
+
+        fs = fsspec.filesystem("https", asynchronous=True, client_kwargs=client_kwargs)
+        session = await fs.set_session()
+        if session is None:
+            print("An error occurred while getting the fsspec session")
+
+        return fs
+
+    def get_https_session(
+        self, bearer_token: bool = False
+    ) -> fsspec.AbstractFileSystem:
+        """
+        Returns a FSSPEC HTTPS session with bearer tokens that are used by CMR.
+        This HTTPS session can be used to download granules if we want to use a direct, lower level API
+        :returns: FSSPEC HTTPFileSystem (aiohttp client session)
+        """
+        req = self.auth.get_session()
+        dummy_oauth_resource = (
+            "https://n5eil01u.ecs.nsidc.org/DP7/ATLAS/ATL06.005/2018.10.14/"
+            "ATL06_20181014045341_02380102_005_01.iso.xml"
+        )
+        resp = req.get(dummy_oauth_resource, allow_redirects=True)
+        if resp.status_code > 400:
+            print("There was an error authorizing this session")
         else:
-            return SessionWithHeaderRedirection(
+            self._authenticated_req_session = req
+        client_kwargs = {
+            "auth": aiohttp.BasicAuth(
                 self.auth._credentials[0], self.auth._credentials[1]
-            )
+            ),
+            "trust_env": True,
+            "cookies": req.cookies.get_dict(),
+        }
+        if bearer_token and self.auth.authenticated:
+            client_kwargs["headers"] = {
+                "Authorization": f'Bearer {self.auth.token["access_token"]}'
+            }
+        session = fsspec.filesystem("https", client_kwargs=client_kwargs)
+        return session
 
-    def open(self, granules: List[Any], provider: str = None) -> List[Any]:
+    @singledispatchmethod
+    def open(
+        self,
+        granules: Union[List[str], List[DataGranule]],
+        provider: str = None,
+    ) -> List[Any]:
         """
-        returns a list of S3 file objects that can be used to access files
-        hosted on S3 by third party libraries like xarray
+        returns a list of fsspec file-like objects that can be used to access files
+        hosted on S3 or HTTPS by third party libraries like xarray.
+
         :param granules: a list of granules(DataGranule) instances or list of files
-        :param provider: cloud provider code, used when the list is made of s3 links
         :returns: a list of s3fs "file pointers" to s3 files
         """
+        print("granules need to be a list of URLs or a list of DataGranule instances")
+        return []
```

Comment:
I typically see an exception raised here, does returning an empty list make sense?

Reply:
Yeah, it should be a `NotImplementedError` exception. Updating it, I usually don't like to rise exceptions but this one is valid.